tests/FunctionalJavascript/TranslationSynchronisationFieldWidgetTest.php

```diff
@@ -0,0 +1,164 @@
+   */
+  protected static $modules = [
+    'system',
+    'node',
+    'user',
+    'field',
+    'text',
+    'options',
+    'oe_translation',
+    'language',
+  ];
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    $this->entityTypeManager = $this->container->get('entity_type.manager');
+
+    $this->entityTypeManager->getStorage('node_type')->create([
+      'name' => 'Page',
+      'type' => 'page',
+    ])->save();
+
+    FieldStorageConfig::create([
+      'field_name' => 'translation_sync',
+      'entity_type' => 'node',
+      'type' => 'oe_translation_translation_sync',
+    ])->save();
+
+    FieldConfig::create([
+      'field_name' => 'translation_sync',
+      'entity_type' => 'node',
+      'bundle' => 'page',
+    ])->save();
+
+    ConfigurableLanguage::create(['id' => 'fr'])->save();
+    ConfigurableLanguage::create(['id' => 'it'])->save();
+
+    $entity_form_display = EntityFormDisplay::collectRenderDisplay(Node::create(['type' => 'page']), 'default');
+    $entity_form_display->setComponent('translation_sync', [
+      'weight' => 1,
+      'region' => 'content',
+      'type' => 'oe_translation_translation_sync_widget',
+      'settings' => [],
+      'third_party_settings' => [],
+    ]);
+    $entity_form_display->save();
+
+    $user = $this->drupalCreateUser([], NULL, TRUE);
+    $this->drupalLogin($user);
+  }
+
+  /**
+   * Tests the Translation Synchronisation field widget.
+   */
+  public function testTranslationSynchronisationFieldWidget(): void {
+    $this->drupalGet('/node/add/page');
+
+    $assert = $this->assertSession();
+    $assert->fieldExists('title[0][value]')->setValue('My page');
+
+    // Assert the select is correct.
+    $select = $assert->selectExists('translation_sync[0][type]');
+    $assert->optionExists('Select type', 'None');
+    $assert->optionExists('Select type', 'Manual');
+    $assert->optionExists('Select type', 'Automatic with minimum threshold');
+    $this->assertEquals('None', $select->find('xpath', "//option[@selected='selected']")->getText());
+
+    // Assert we don't see a configuration fieldset.
+    $this->assertFalse($this->getSession()->getPage()->find('css', '#edit-translation-sync-0-configuration')->isVisible());
+
+    // Save the node with no value.
+    $this->getSession()->getPage()->pressButton('Save');
+    /** @var \Drupal\node\NodeInterface $node */
+    $node = $this->entityTypeManager->getStorage('node')->load(1);
+    $this->assertTrue($node->get('translation_sync')->isEmpty());
+    $this->assertEmpty($node->get('translation_sync')->getValue());
+
+    // Edit the node again and configure it to be Manual.
+    $this->drupalGet($node->toUrl('edit-form'));
+    $this->getSession()->getPage()->selectFieldOption('Select type', 'manual');
+    $this->getSession()->wait(10);
```

Comment:
Not a huge fan of adding this wait here. I understand you want to make sure the configuration is not loaded but I wonder if it is even necessary?

Reply:
No, the wait here is to give enough time for the #states to actually hide or show. We wanna make sure that we dont have a false positive and in fact the states does not show anything. Where the states actually shows something, we do a wait for element instead of a random amount of time.

Reply:
If we wanted to be super precise we could add a condition to the wait so it ends if it finds the container, but probably not worh the effort.